packages/grafana-runtime/src/services/templateSrv.ts

```diff
@@ -16,7 +34,12 @@ export interface TemplateSrv {
   /**
    * Replace the values within the target string.  See also {@link InterpolateFunction}
    */
-  replace(target?: string, scopedVars?: ScopedVars, format?: string | Function): string;
+  replace(
+    target?: string,
+    scopedVars?: ScopedVars,
+    format?: string | Function,
+    interpolations?: VariableInterpolation[]
```

Comment:
Interesting idea with recording the interpolations. I don't think we often use functions that mutate arguments, so I suggest providing a bold doc comment about why and how to use it.

Reply:
@dprokop yea, agree a good doc comment here would be good. 

or we add another function that returns `{ result: string, interpolations: VariableInterpolation[] }`, but not sure what it should be named, and adds more duplication/choice

Reply:
Added a comment in 73cf2e9b26cceb27cd893e6c3b5c71be3c2d790c

Reply:
@torkelo / @dprokop - could you have another look? Does it look good?